app/src/app/components/ErrorNotification.tsx

```diff
@@ -0,0 +1,78 @@
+"use client"
+import {useEffect, useState} from 'react';
+import {useMapStore} from '../store/mapStore';
+import {AlertDialog, Button, Flex, Text} from '@radix-ui/themes';
+import * as Toast from '@radix-ui/react-toast';
+
+export const ErrorNotification = () => {
+  const errorNotification = useMapStore(state => state.errorNotification);
+  const [errorUiActive, setErrorUiActive] = useState(true);
+
+  useEffect(() => {
+    errorNotification.message && setErrorUiActive(true);
+  }, [errorNotification]);
+
+  if (!errorUiActive || !errorNotification.message || !errorNotification.severity) {
+    return null;
+  }
+  const ErrorDescription = () => (
+    <>
+      <Text as="p" mb={'4'}>
+        {errorNotification.message}
+      </Text>
+      <Text as="p" mb={'4'}>
+        {!!errorNotification?.id && <i>Error ID: {errorNotification.id}</i>}
+      </Text>
+      <Text as="p" mb={'4'}>
+        {!!errorNotification?.severity && <i>Severity: {errorNotification.severity}</i>}
+      </Text>
```

Comment:
PP: Not sure we need the severity here.

Q: Do you know if Sentry will capture these errors? It would be good to still have them tracked there but not pressing to figure that out

Reply:
I'm using severity as a way to sort alert type - dialog, toast, silent log ?

Reply:
I'm not super familiar with sentry but I'm sure this can integrate with it

Reply:
Makes sense – just not sure it needs to be exposed to the user!